jmix-pivottable/pivottable-flowui/src/main/java/io/jmix/pivottableflowui/component/PivotTable.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.pivottableflowui.component;
+
+import com.vaadin.flow.function.SerializableConsumer;
+import elemental.json.JsonObject;
+import elemental.json.JsonValue;
+import io.jmix.core.Messages;
+import io.jmix.core.metamodel.datatype.FormatStrings;
+import io.jmix.core.metamodel.datatype.FormatStringsRegistry;
+import io.jmix.core.security.CurrentAuthentication;
+import io.jmix.pivottableflowui.export.model.PivotData;
+import io.jmix.pivottableflowui.kit.component.JmixPivotTable;
+import io.jmix.pivottableflowui.kit.component.model.AggregationMode;
+import io.jmix.pivottableflowui.kit.component.model.Renderer;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import java.text.DecimalFormatSymbols;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Integration of an open-source Javascript Pivot Table written by <a href="https://nicolas.kruchten.com/">Nicolas Kruchten</a>.
+ * A pivot table component is a powerful data analysis tool that allows users to summarize, analyze and present
+ * datasets in a flexible and interactive manner.
+ * @see <a href="https://github.com/nicolaskruchten/pivottable/wiki">Pivot table documentation</a>
+ */
+public class PivotTable<T> extends JmixPivotTable<T> implements ApplicationContextAware, InitializingBean {
+
+    protected ApplicationContext applicationContext;
+    protected Messages messages;
+    protected FormatStringsRegistry formatStringsRegistry;
+    protected CurrentAuthentication currentAuthentication;
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void afterPropertiesSet() {
+        autowireDependencies();
+
+        initLocalization();
+    }
+
+    public void requestPivotData(String dateTimeParseFormat, String dateParseFormat, String timeParseFormat,
```

Comment:
1. public API must have JavaDoc
2. It looks like an internal API. At the moment, I don't see any reason to have this API directly in the pivot. You can access its Element and invoke the JS function. Or you can create a static helper class for it.